docs/source/tensor_view.rst

```diff
@@ -0,0 +1,74 @@
+.. currentmodule:: torch
+
+.. _tensor-view-doc:
+
+Tensor Views
+=================
+
+`Tensor` and `Storage` are two core concepts in PyTorch. `Tensor` is the main user facing data structure,
+while `Storage` is mostly internal. Users usually don't have to deal with `Storage` directly.
+
+`Tensor` contains data, as well as its metadata such as size, dtype and device. But `Tensor` doesn't
+hold the physical data. Instead it is built on top of `Storage`, which is a data structure that
+actually holds physical data. In other words, there's always a storage backing up a tensor.
+
+PyTorch has this layered `Tensor` - `Storage` structure mainly to allow a tensor to be `View` of an existing tensor
+where they share the same underlying storage. Supporting `View` allows us to do fast and memory efficient
+reshaping, slicing and element-wise operations.
+
+For example, to get a view of an existing tensor `t`, you can call `t.view(...)`.
+
+::
+
+    >>> t = torch.rand(4, 4)
+    >>> b = t.view(2, 8)
+    >>> t.storage().data_ptr() == b.storage().data_ptr()  # `t` and `b` share the same storage.
+    True
+    # Modifying views will change the original tensor as well.
+    >>> b[0][0] = 3.14
+    >>> t[0][0]
+    tensor(3.14)
+
+Note that since views share underlying storage with the original tensor, if you edit the data
+in the view, it will be reflected in the original tensor as well.
+
+Other than :meth:`~torch.Tensor.view` shown above, there're more PyTorch ops taking advantange of efficiency of views.
+For reference, here’s a list of all ops in PyTorch that return view of input tensors:
+
+- Basic indexing return views.
+
+  When accessing the contents of a tensor via indexing, PyTorch follows Numpy behaviors
+  that basic indexing returns views, while advanced indexing returns a copy.
+  Assignment with both basic and advanced indexing is in-place. See more examples in
+  `Numpy indexing documentation <https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html>`.
+- :meth:`~torch.Tensor.transpose`
+- :meth:`~torch.Tensor.as_strided`
+- :meth:`~torch.Tensor.diagonal`
+- :meth:`~torch.Tensor.expand`
+- :meth:`~torch.Tensor.narrow`
+- :meth:`~torch.Tensor.permute`
+- :meth:`~torch.Tensor.select`
+- :meth:`~torch.Tensor.squeeze`
+- :meth:`~torch.Tensor.t`
```

Comment:
nit `tensor.T` (numpy style transposition) returns a view as well.

Reply:
@albanD Did we document this method somewhere? ;)
I can write text but curious if I'm missing it

Reply:
Yes, it is there: https://pytorch.org/docs/stable/tensors.html#torch.Tensor.T
Even though it looks weird in the doc :D